ansible_base/rbac/api/serializers.py

```diff
@@ -0,0 +1,232 @@
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import transaction
+from django.db.utils import IntegrityError
+from django.utils.translation import gettext_lazy as _
+from rest_framework import serializers
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.fields import flatten_choices_dict, to_choices_dict
+from rest_framework.serializers import ValidationError
+
+from ansible_base.lib.abstract_models.common import get_url_for_object
+from ansible_base.lib.serializers.common import CommonModelSerializer
+from ansible_base.lib.utils.validation import ansible_id_validator
+from ansible_base.rbac.models import RoleDefinition, RoleTeamAssignment, RoleUserAssignment
+from ansible_base.rbac.permission_registry import permission_registry  # careful for circular imports
+from ansible_base.rbac.validators import validate_permissions_for_model
+
+
+class ChoiceLikeMixin(serializers.ChoiceField):
+    """
+    This uses a ForeignKey to populate the choices of a choice field.
+    This also manages some string manipulation, right now, adding the local service name.
+    """
+
+    default_error_messages = serializers.PrimaryKeyRelatedField.default_error_messages
+    psuedo_field = None  # define in subclass
```

Comment:
pseudo

Reply:
Also not used by anything?

Reply:
ah you're right, I used it for grabbing the field help_text and for some reason or another it didn't work and something else was done.